scrapy/core/downloader/__init__.py

```diff
@@ -99,9 +103,14 @@ def needs_backout(self):
     def _get_slot(self, request, spider):
         key = self._get_slot_key(request, spider)
         if key not in self.slots:
-            conc = self.ip_concurrency if self.ip_concurrency else self.domain_concurrency
+            conc = self.per_slot_settings.get(key, {}).get('concurrency', self.ip_concurrency if self.ip_concurrency else self.domain_concurrency)
             conc, delay = _get_concurrency_delay(conc, spider, self.settings)
-            self.slots[key] = Slot(conc, delay, self.randomize_delay)
+            delay = self.per_slot_settings.get(key, {}).get('delay', delay)
+            randomize_delay = self.per_slot_settings.get(key, {}).get('randomize_delay', self.randomize_delay)
+            new_slot = Slot(conc, delay, randomize_delay)
+            self.slots[key] = new_slot
+            logger.debug(
+                f"Downloader slot '{key}' - created {'(from per slot settings)' if key in self.per_slot_settings.keys() else ''}: {new_slot}")
```

Comment:
I understand this may have helped while implementing this feature, but I am not sure we should keep these debug messages around.

Same for the message below.